modules/gltf/gltf_document.cpp

```diff
@@ -3078,15 +3078,18 @@ Ref<Image> GLTFDocument::_parse_image_bytes_into_image(Ref<GLTFState> p_state, c
 		Error err = ext->parse_image_data(p_state, p_bytes, p_mime_type, r_image);
 		ERR_CONTINUE_MSG(err != OK, "GLTF: Encountered error " + itos(err) + " when parsing image " + itos(p_index) + " in file " + p_state->filename + ". Continuing.");
 		if (!r_image->is_empty()) {
+			r_file_extension = ext->get_image_file_extension();
```

Comment:
doesn't this need to pass in `p_index` ? Or is it expected that a given extension only handles exactly one image format.

Reply:
The code expects that one GLTFDocumentExtension handles importing one kind of image format (can be "multiple formats" with one file extension like Lossy WebP and Lossless WebP). If multiple file extensions are needed, it's not a big deal for a plugin/add-on to supply multiple GLTFDocumentExtension classes. In fact that's probably for the best since it would keep the code abstracted. Note that the import code checks each image individually so it may be the case that multiple GLTFDocumentExtensions are used to import one glTF file.